src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageProcessingConnector.cs

```diff
@@ -76,9 +75,10 @@ static void ConvertToPendingOperations(OutboxMessage deduplicationEntry, Pending
             }
         }
 
-        static List<TransportOperation> ConvertToOutboxOperations(IEnumerable<Transports.TransportOperation> operations)
+        static TransportOperation[] ConvertToOutboxOperations(Transports.TransportOperation[] operations)
```

Comment:
Out of scope for this PR, but it would be nice if we could just use the same class here (and prevent the extra allocation), or rename one to make it clearer as to what's going on!